### ISSUE TITLE
[inductor] SDPA fusion silently lost when a SymFloat scale is tensorified (dynamic shapes, e.g. head_dim ** 0.5)

### ISSUE THREAD

**Issue by HussainNizamani** [pytorch/pytorch#194444]
### 🐛 Describe the bug

Under `torch.compile` with dynamic shapes, writing SDPA with the canonical scale

```python
scale = head_dim ** 0.5
```

produces a SymFloat scale that `tensorify_python_scalars` converts into a 0-d tensor (`aten.scalar_tensor.default`) before the joint-graph pattern passes run (`torch/fx/passes/_tensorify_python_scalars.py:179-205`). The SDPA fusion guard rejects non-scalar scales — necessary since #191203, where such a 0-d tensor scale reached `aten::scaled_dot_product_attention`'s `float? scale` argument and crashed — so these graphs silently fall back to unfused attention instead of fusing.

Trading the crash for a silent fusion loss is the safe immediate behavior, but it is a real regression in kernel selection for the most idiomatic dynamic-shape attention. The scalar should be recoverable: the sympy expression is still reachable from the `scalar_tensor` node created by the pass, so the guard (or the replacement construction) could recover it and keep the fusion.

Repro (fusion counter stays 0; on builds without the #191496 guard this crashes like #191203 instead):

```python
import torch
from torch._dynamo.utils import counters

def model(q, k, v):
    scale = q.size(0) ** 0.5
    return torch.nn.functional.softmax(
        q @ k.transpose(-2, -1) / scale, dim=-1) @ v

args = [torch.randn(8, 4, 4, 4) for _ in range(3)]
torch.compile(model, fullgraph=True, dynamic=True)(*args)
print(counters["inductor"]["fuse_attention"])  # 0 — fusion skipped
```

TODOs pointing at this issue: `torch/_inductor/fx_passes/fuse_attention.py` (`_is_supported_scale`) and `test/inductor/test_fused_attention.py` (`_test_pattern_fails_with_symfloat_scale`), both added in #191496.

### Versions

main (also reproduces on recent nightlies)

{AI-Assisted}

cc @chauhang @penguinwu @ezyang @bobrenjc93 @aditvenk @laithsakka @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @aakhundov @coconutruben @jataylo @drisspg @liangel-02 @howardzhang-cv

**Comment by DreamingWater** [comment]
Hi, I'd like to work on this issue. I plan to first reproduce it on current main/nightly, then investigate recovering only tensorified symbolic scalar scales while preserving the rejection of genuine tensor scales introduced in #191496, and add focused regression tests.Hi, I'd like to work on this issue. I plan to first reproduce it on current main/nightly, then investigate recovering only tensorified symbolic scalar scales while preserving the rejection of genuine tensor scales introduced in #191496, and add focused regression tests.

I couldn't find an open PR or linked branch addressing #194444. Is anyone already working on it? If not, could a maintainer confirm that this is available and mark it actionable?

I couldn't find an open PR or linked branch addressing #194444. Is anyone already working on it? If not, could a maintainer confirm that this is available and mark it actionable?
